File: mmupnp/src/test/java/net/mm2d/upnp/empty/EmptyDeviceTest.java
/*
 * Copyright (c) 2018 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.upnp.empty;

import net.mm2d.upnp.Device;
import net.mm2d.upnp.HttpClient;
import net.mm2d.upnp.IconFilter;
import net.mm2d.upnp.SsdpMessage;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import static org.hamcrest.Matchers.*;
import static org.junit.Assert.*;
import static org.mockito.Mockito.*;

@SuppressWarnings("NonAsciiCharacters")
@RunWith(JUnit4.class)
public class EmptyDeviceTest {

    @Test
    public void loadIconBinary() {
        final Device device = new EmptyDevice();
        device.loadIconBinary(mock(HttpClient.class), mock(IconFilter.class));
    }

    @Test
    public void getControlPoint() {
        final Device device = new EmptyDevice();
        assertThat(device.getControlPoint(), is(not(nullValue())));
    }

    @Test
    public void updateSsdpMessage() {
        final Device device = new EmptyDevice();
        device.updateSsdpMessage(mock(SsdpMessage.class));
    }

    @Test
    public void getSsdpMessage() {
        final Device device = new EmptyDevice();
        assertThat(device.getSsdpMessage(), is(not(nullValue())));
    }

    @Test
    public void getExpireTime() {
        final Device device = new EmptyDevice();
        assertThat(device.getExpireTime() >= 0, is(true));
    }

    @Test
    public void getDescription() {
        final Device device = new EmptyDevice();
        assertThat(device.getDescription(), is(not(nullValue())));
    }

    @Test
    public void getScopeId() {
        final Device device = new EmptyDevice();
        assertThat(device.getScopeId(), is(0));
    }

    @Test
    public void getValue() {
        final Device device = new EmptyDevice();
        assertThat(device.getValue(""), is(nullValue()));
    }

    @Test
    public void getValueWithNamespace() {
        final Device device = new EmptyDevice();
        assertThat(device.getValueWithNamespace("", ""), is(nullValue()));
    }

    @Test
    public void getLocation() {
        final Device device = new EmptyDevice();
        assertThat(device.getLocation(), is(not(nullValue())));
    }

    @Test
    public void getBaseUrl() {
        final Device device = new EmptyDevice();
        assertThat(device.getBaseUrl(), is(not(nullValue())));
    }

    @Test
    public void getIpAddress() {
        final Device device = new EmptyDevice();
        assertThat(device.getIpAddress(), is(not(nullValue())));
    }

    @Test
    public void getUdn() {
        final Device device = new EmptyDevice();
        assertThat(device.getUdn(), is(not(nullValue())));
    }

    @Test
    public void getUpc() {
        final Device device = new EmptyDevice();
        assertThat(device.getUpc(), is(nullValue()));
    }

    @Test
    public void getDeviceType() {
        final Device device = new EmptyDevice();
        assertThat(device.getDeviceType(), is(not(nullValue())));
    }

    @Test
    public void getFriendlyName() {
        final Device device = new EmptyDevice();
        assertThat(device.getFriendlyName(), is(not(nullValue())));
    }

    @Test
    public void getManufacture() {
        final Device device = new EmptyDevice();
        assertThat(device.getManufacture(), is(nullValue()));
    }

    @Test
    public void getManufactureUrl() {
        final Device device = new EmptyDevice();
        assertThat(device.getManufactureUrl(), is(nullValue()));
    }

    @Test
    public void getModelName() {
        final Device device = new EmptyDevice();
        assertThat(device.getModelName(), is(not(nullValue())));
    }

    @Test
    public void getModelUrl() {
        final Device device = new EmptyDevice();
        assertThat(device.getModelUrl(), is(nullValue()));
    }

    @Test
    public void getModelDescription() {
        final Device device = new EmptyDevice();
        assertThat(device.getModelDescription(), is(nullValue()));
    }

    @Test
    public void getModelNumber() {
        final Device device = new EmptyDevice();
        assertThat(device.getModelNumber(), is(nullValue()));
    }

    @Test
    public void getSerialNumber() {
        final Device device = new EmptyDevice();
        assertThat(device.getSerialNumber(), is(nullValue()));
    }

    @Test
    public void getPresentationUrl() {
        final Device device = new EmptyDevice();
        assertThat(device.getPresentationUrl(), is(nullValue()));
    }

    @Test
    public void getIconList() {
        final Device device = new EmptyDevice();
        assertThat(device.getIconList(), is(not(nullValue())));
    }

    @Test
    public void getServiceList() {
        final Device device = new EmptyDevice();
        assertThat(device.getServiceList(), is(not(nullValue())));
    }

    @Test
    public void findServiceById() {
        final Device device = new EmptyDevice();
        assertThat(device.findServiceById(""), is(nullValue()));
    }

    @Test
    public void findServiceByType() {
        final Device device = new EmptyDevice();
        assertThat(device.findServiceByType(""), is(nullValue()));
    }

    @Test
    public void findAction() {
        final Device device = new EmptyDevice();
        assertThat(device.findAction(""), is(nullValue()));
    }

    @Test
    public void isEmbeddedDevice() {
        final Device device = new EmptyDevice();
        assertThat(device.isEmbeddedDevice(), is(false));
    }

    @Test
    public void getParent() {
        final Device device = new EmptyDevice();
        assertThat(device.getParent(), is(nullValue()));
    }

    @Test
    public void getDeviceList() {
        final Device device = new EmptyDevice();
        assertThat(device.getDeviceList(), is(not(nullValue())));
    }

    @Test
    public void findDeviceByType() {
        final Device device = new EmptyDevice();
        assertThat(device.findDeviceByType(""), is(nullValue()));
    }

    @Test
    public void findDeviceByTypeRecursively() {
        final Device device = new EmptyDevice();
        assertThat(device.findDeviceByTypeRecursively(""), is(nullValue()));
    }

    @Test
    public void isPinned() {
        final Device device = new EmptyDevice();
        assertThat(device.isPinned(), is(false));
    }
}

File: mmupnp/src/test/java/net/mm2d/upnp/empty/EmptyDeviceTest.kt
/*
 * Copyright (c) 2018 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.upnp.empty

import com.google.common.truth.Truth.assertThat
import io.mockk.mockk
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4

@RunWith(JUnit4::class)
class EmptyDeviceTest {

    @Test
    fun loadIconBinary() {
        val device = EmptyDevice
        device.loadIconBinary(mockk()) { it }
    }

    @Test
    fun getControlPoint() {
        val device = EmptyDevice
        assertThat(device.controlPoint).isNotNull()
    }

    @Test
    fun updateSsdpMessage() {
        val device = EmptyDevice
        device.updateSsdpMessage(mockk())
    }

    @Test
    fun getSsdpMessage() {
        val device = EmptyDevice
        assertThat(device.ssdpMessage).isNotNull()
    }

    @Test
    fun getExpireTime() {
        val device = EmptyDevice
        assertThat(device.expireTime).isAtLeast(0)
    }

    @Test
    fun getDescription() {
        val device = EmptyDevice
        assertThat(device.description).isNotNull()
    }

    @Test
    fun getScopeId() {
        val device = EmptyDevice
        assertThat(device.scopeId).isEqualTo(0)
    }

    @Test
    fun getValue() {
        val device = EmptyDevice
        assertThat(device.getValue("")).isNull()
    }

    @Test
    fun getValueWithNamespace() {
        val device = EmptyDevice
        assertThat(device.getValueWithNamespace("", "")).isNull()
    }

    @Test
    fun getLocation() {
        val device = EmptyDevice
        assertThat(device.location).isNotNull()
    }

    @Test
    fun getBaseUrl() {
        val device = EmptyDevice
        assertThat(device.baseUrl).isNotNull()
    }

    @Test
    fun getIpAddress() {
        val device = EmptyDevice
        assertThat(device.ipAddress).isNotNull()
    }

    @Test
    fun getUdn() {
        val device = EmptyDevice
        assertThat(device.udn).isNotNull()
    }

    @Test
    fun getUpc() {
        val device = EmptyDevice
        assertThat(device.upc).isNull()
    }

    @Test
    fun getDeviceType() {
        val device = EmptyDevice
        assertThat(device.deviceType).isNotNull()
    }

    @Test
    fun getFriendlyName() {
        val device = EmptyDevice
        assertThat(device.friendlyName).isNotNull()
    }

    @Test
    fun getManufacture() {
        val device = EmptyDevice
        assertThat(device.manufacture).isNull()
    }

    @Test
    fun getManufactureUrl() {
        val device = EmptyDevice
        assertThat(device.manufactureUrl).isNull()
    }

    @Test
    fun getModelName() {
        val device = EmptyDevice
        assertThat(device.modelName).isNotNull()
    }

    @Test
    fun getModelUrl() {
        val device = EmptyDevice
        assertThat(device.modelUrl).isNull()
    }

    @Test
    fun getModelDescription() {
        val device = EmptyDevice
        assertThat(device.modelDescription).isNull()
    }

    @Test
    fun getModelNumber() {
        val device = EmptyDevice
        assertThat(device.modelNumber).isNull()
    }

    @Test
    fun getSerialNumber() {
        val device = EmptyDevice
        assertThat(device.serialNumber).isNull()
    }

    @Test
    fun getPresentationUrl() {
        val device = EmptyDevice
        assertThat(device.presentationUrl).isNull()
    }

    @Test
    fun getIconList() {
        val device = EmptyDevice
        assertThat(device.iconList).isNotNull()
    }

    @Test
    fun getServiceList() {
        val device = EmptyDevice
        assertThat(device.serviceList).isNotNull()
    }

    @Test
    fun findServiceById() {
        val device = EmptyDevice
        assertThat(device.findServiceById("")).isNull()
    }

    @Test
    fun findServiceByType() {
        val device = EmptyDevice
        assertThat(device.findServiceByType("")).isNull()
    }

    @Test
    fun findAction() {
        val device = EmptyDevice
        assertThat(device.findAction("")).isNull()
    }

    @Test
    fun isEmbeddedDevice() {
        val device = EmptyDevice
        assertThat(device.isEmbeddedDevice).isFalse()
    }

    @Test
    fun getParent() {
        val device = EmptyDevice
        assertThat(device.parent).isNull()
    }

    @Test
    fun getDeviceList() {
        val device = EmptyDevice
        assertThat(device.deviceList).isNotNull()
    }

    @Test
    fun findDeviceByType() {
        val device = EmptyDevice
        assertThat(device.findDeviceByType("")).isNull()
    }

    @Test
    fun findDeviceByTypeRecursively() {
        val device = EmptyDevice
        assertThat(device.findDeviceByTypeRecursively("")).isNull()
    }

    @Test
    fun isPinned() {
        val device = EmptyDevice
        assertThat(device.isPinned).isFalse()
    }
}
